src/linagora.esn.contact/app/app.config.js

```diff
@@ -0,0 +1,10 @@
+angular.module('linagora.esn.contact')
+  .config(registerI18N);
+
+function registerI18N($translateProvider) {
+  $translateProvider.translations('en', require('../i18n/en.json'));
```

Comment:
I know that @MichaelBailly told to do it like this but we also discussed about how to do it with webpack which will allow us to avoid having to update all the SPAs when adding a locale support:

Webpack can concatenates all the files it finds in a folder and generate a single one, then in this file we do require just only the generated file. WDYT about this?

Reply:
I agree with you. But do you have in your mind how we apply it? Because we need to register locale for $translateProvider, how do you automate it? Read locales from file names?

Reply:
A folder containing fr.json, en.json, etc can be translated to a single file like

```json
{
"fr": {}
"en": {}
}
```

And then we can have a registerI18N iterating in the content of this single file by key

Reply:
Or we store the supported locales for OP as a configuration at backend/

Reply:
:+1: 

But the combined JSON file can be really long.

Reply:
As long as the sum of the locales

Reply:
umh, thinking about another solution now. We may be able to generate this angularjs block at compile time also based on the content of a folder. Then the day we add a new file in the folder, the block is updated at build time: The files are remaining the same and we do not have to change anything in the code, webpack is doing it for us.
The day we can lazy load locale files, we will just have to change things in a single place. 

WDYT?

Reply:
Sound nice! 

I :heart:  the sentence: `webpack is doing it for us`

But I need to check how we do it. Please tell me you have an idea (just for save time)

Reply:
a webpack plugin? Check this one https://github.com/linagora/openpaas-dashboard/blob/master/vue.config.js#L8

Reply:
@chamerling with your solution, the dynamic locales issue is not absolutely resolved. When you need to add one more locale, you still need to add the locale file in the i18n folder. So, adding 1 more line like this is not a big deal.

Reply:
If the problem we're trying to avoid is having to "duplicate" the changes in every SPA repository, I think the only way to do that is to move all the translation files of the SPAs into a single repository (i.e. `esn-frontend-common-libs`). Otherwise, every solution is half-baked and only introduces more complexity.

Considering that adding a new language is something we do once in a blue moon, it is not worth doing all of this imo.

Reply:
OK so @tuanlc, let's keep it like it is. This is not a big feature and will not move for years.